### ISSUE TITLE
[JENKINS-29198] Console logs get truncated without reason

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#749]
<p>I have a bug which looks similar to <a href="https://issues.jenkins.io/browse/JENKINS-14899" title="consoleText stops updating at 5000 lines until end of build job" class="issue-link" data-issue-key="JENKINS-14899"><del>JENKINS-14899</del></a>:<br/>
Sometimes, when a job runs for a long time without anything getting output, the jenkins page stops updating until the job finishes. Then it add the completion lines (SUCCESS or FAILURE) to the job.<br/>
The difference is that the number of line does not matter (last bug occurence was on a job that output 264 lines (+ an unknown number of non-output lines)</p>

<p>This results in a <b>truncated /console, /consoleFull and /consoleText</b>.</p>

<p>All files do end with the two lines (if it succeeds):<br/>
    Notifying upstream projects of job completion<br/>
    Finished: SUCCESS<br/>
but hundreds of lines didn't get stored by jenkins (over 10 minutes of logs for my last job!)</p>

<p>I can often see this bug when I'm running a long impala query, where the output stops before the status line (something like "Returned 0 row(s) in 1987.42s") and the rest of the console logs never get output.</p>

<p>This happens approximately one time out of 20 for the jobs I'm running myself on this Jenkins server.</p>

---
<details><summary><i>Originally reported by <a title="Namikaze Minato" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=camusensei">camusensei</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-29198" target="_blank">Console logs get truncated without reason</a></i></summary>
<i><ul>
<li><b>reported by</b>: camusensei
<li><b>status</b>: Open
<li><b>priority</b>: Blocker
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 20
<li><b>watchers</b>: 22
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck">danielbeck</a>:</i>
<p>Please make sure any issues you experience happen on a reasonably recent version of Jenkins. Since it's been more than a year since 1.553 was released, this issue may well be obsolete.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rjfenton">rjfenton</a>:</i>
<p>As of today am also seeing this on version 2.154 I don't see any success or failure messages. I also noticed that loading output for nodes in the blue ocean view seems to give a red console error. (I'm guessing these are somehow loaded from the log which doesn't have the info since it's truncated.)</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=rizwan_inamdar">rizwan_inamdar</a>:</i>
<p>Please increase the priority.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=laiph">laiph</a>:</i>
<p>Indeed. For us, this happens when we run like 6 builds in parallel. However, if we break them in two, having only 3 builds run at a time, same setup, just two sequence of 3 builds, we get full logs.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=elliot_nelson">elliot_nelson</a>:</i>
<p>Recently seen this a couple times on Jenkins 2.193.</p>

<p> </p>

<p>In our case, it's a multijob that ends like this:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">...
Finished Build : #18084 of Job : fvt-worker-5 with status : SUCCESS
Finished Build : #18084 of Job : fvt-worker-20 with status : SUCCESS
Finished Build : #18085 of Job : fvt-worker-7 with status : SUCCESS
Finished Build : #18085 of Job : fvt-worker-23 with status : SUCCESS
Finished Build : #18084 of Job : fvt-worker-8 with status : SUCCESS
Finished Build : Finished Build :  </pre>
</div></div>
<p> </p>

<p>It's clear the job finished successfully (including another dozen subjobs that should be listed but aren't), and that it also completed all the publisher steps as expected, but there's about 120 lines of log at the end there that are missing (affects /console, /consoleFull, and /consoleText).</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=akmjenkins">akmjenkins</a>:</i>
<p>In our case - "Console logs get truncated when job runs for a long time". Are there any log size limit !?</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=akmjenkins">akmjenkins</a>:</i>
<p>We are using core 2.164.3 version. Ay idea, by upgrading to any specific version - will it help !! </p>

<p>And/or any workaround we can try !!</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=alok15ee">alok15ee</a>:</i>
<p>Did anyone get a workaround for this? </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=camusensei">camusensei</a>:</i>
<p>Few Jenkins versions have this bug, just upgrade.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=alok15ee">alok15ee</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=camusensei" class="user-hover" rel="camusensei">camusensei</a> What version do you recommend? I have tried with 2.204.1 and 2.222.3 and I still run into the same issue.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=camusensei">camusensei</a>:</i>
<p>I have no problem on my jenkins servers:</p>

<p>2.249.1<br/>
 2.138<br/>
 2.27<br/>
 2.26<br/>
 2.204.5<br/>
 2.235.5<br/>
 2.138.1</p>

<p>You could try to test if the problem is reproducible, like a loop printing messages, e.g.:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>#!/bin/bash
while sleep 1; do printf '%*s\n' 300 "$SECONDS"; done
</pre>
</div></div>
<p>Try it and see if your problem always happens after the same time, or if such a simple script does not allow you to reproduce the problem?</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=alok15ee">alok15ee</a>:</i>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=camusensei" class="user-hover" rel="camusensei">camusensei</a> Nope it didn't work either. I think it also depends on the amount of data in one line we write to the BufferedInputStream which is causing this issue to pop up.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=camusensei">camusensei</a>:</i>
<p>I think you need to give more information on your problem which seems to be different from the one I used to have. Maybe open a new issue, and describe preciseley anything you can.</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=JIRAUSER140876">JIRAUSER140876</a>:</i>
<p>I am seeing this issue right now on Jenkins 2.335!</p>

<p>There's something gone awry.</p>

<p>We use kubernetes and had been having some connection issues we solved by stopping the use of websockets to agents, but the truncated logs keep happening, developers cannot debug builds, and I cannot tell them why it's failed or why there are no logs...</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=JIRAUSER140876">JIRAUSER140876</a>:</i>
<p><tt>The only thing interesting I see in system jenkins logs is:</tt></p>

<p><tt>Mar 04, 2022 4:56:35 PM WARNING jenkins.branch.WorkspaceLocatorImpl getWorkspaceRoot</tt><tt><a href="https://issues.jenkins.io/browse/JENKINS-2111" title="removing a job (including multibranch/org folder branches/repos) does not remove the workspace" class="issue-link" data-issue-key="JENKINS-2111"><del>JENKINS-2111</del></a> path sanitization ineffective when using legacy Workspace Root Directory ‘${ITEM_ROOTDIR}/workspace’; switch to ‘${JENKINS_HOME}/workspace/${ITEM_FULL_NAME}’ as in <a href="https://issues.jenkins.io/browse/JENKINS-8446" title="&#39;builds&#39; directory in a separate location?" class="issue-link" data-issue-key="JENKINS-8446"><del>JENKINS-8446</del></a> / <a href="https://issues.jenkins.io/browse/JENKINS-21942" title="Unreliable defaulting of Jenkins.workspaceDir" class="issue-link" data-issue-key="JENKINS-21942">JENKINS-21942</a></tt></p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=JIRAUSER153600">JIRAUSER153600</a>:</i>
<p>Hello,</p>

<p>I'm affected by this issue too, it's really annoying as it forces user to attempt a local rebuild to obtain logs, which even worse, doesn't work when the issue is specific to the server...</p>

<p>I'm reproducing the issue with the following pipeline job:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-groovy">
pipeline {
    agent any

    stages {
        stage(<span class="code-quote">'Hello'</span>) {
            matrix {
                axes {
                    axis {
                        name <span class="code-quote">'blarg'</span>
                        values <span class="code-quote">'foo'</span>, <span class="code-quote">'bar'</span>, <span class="code-quote">'baz'</span>, <span class="code-quote">'qux'</span>, <span class="code-quote">'wizz'</span>
                    }
                }
                stages {
                    stage(<span class="code-quote">'Main'</span>) {
                        steps {
                            sh """
                            <span class="code-keyword">for</span> i <span class="code-keyword">in</span> \$(seq 1 ${TURNS}); <span class="code-keyword">do</span>
                                echo hello world \${i}!
                            done
                            """
                        }
                    }
                }
            }
        }
    }
}
</pre>
</div></div>
<p>When <b>TURNS</b> is 100000, all logs look fine, when set to 200000, the logs are truncated and the job takes 10 to 40 times longer to complete.<br/>
Note that when I redirect the logs to files by doing for example:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-sh">
(<span class="code-object">for</span> i <span class="code-object">in</span> \$(seq 1 ${TURNS}); <span class="code-object">do</span>
    echo hello world \${i}!
<span class="code-object">done</span>) | tee ${blarg}.log
</pre>
</div></div>
<p>Log files created are complete, so it doesn't look like a disk I/O bottleneck issue.</p>

<p> </p>

<p>Jenkins version: 2.375.1</p>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=JIRAUSER153600">JIRAUSER153600</a>:</i>
<p>In my situation, I think that this issue is related to <a href="https://issues.jenkins.io/browse/JENKINS-56446" class="external-link" rel="nofollow">https://issues.jenkins.io/browse/JENKINS-56446</a></p>
